samples/net/sockets/http_client/boards/serpente.overlay

```diff
@@ -0,0 +1,12 @@
+/*
```

Comment:
How do you use networking and http_client with this board?

Reply:
You are right, this overlay file should not be needed. I just deleted it locally and ran the test suite `./scripts/sanitycheck -p serpente`. This also runs "samples/net/sockets/http_client/sample.net.sockets.http_client" which fails with `/home/alexander/Documents/Git/private/zephyr/zephyr/drivers/serial/uart_sam0.c:1031:37: error: 'uart_sam0_driver_api' defined but not used [-Werror=unused-const-variable=]` because no sercom is configured for uart by default.
Why does this sample even get compiled for the serpente board? The serpente.yml does not specify any of the tags from the sample, if I understand correctly. What did I miss?

Reply:
> Why does this sample even get compiled for the serpente board?

Yep, that is the issue here. @nashif any idea?

Reply:
Looks like this is a misunderstanding on my side. `sanitycheck` intersects the platform.supported and the testcase.depends_on sets to calculate which testcases should run for which platform. However, the `http_client/sample.yaml` does not have any depends_on definition (this is probably the actual bug here? And maybe the reason for some of the other tests to fail as well.)

Reply:
Do you want me to add a `depends_on: netif` to the `http_client/sample.yaml` in this PR or as a separate PR? Or would some other change be better at all?

Reply:
I wonder why the missing `depends_on: netif` is not causing issues with other boards that do not have networking support. Also not all networking samples have dependency mentioned in the `sample.yaml` file. But if that depends_on fixes this issue, you can certainly add a commit to this PR.

Reply:
I've been wondering the same, but my understanding of the whole build process is not that good yet. I'd be interested in an answer too.
Alright, I'll amend the PR and add the depends_on.